File: objects/waveclass_unittest/MATLAB/testconfirm.m
clc;clear all; close all;
wType = 'etaImport';
bemFreq = [5.19999512307279,0.0199999977946844];
wDepth = 'infinite';
rampTime = 100;
dt = 0.1;
maxIt = 4000;
g = 9.81;
rho = 1000;
endTime = 400;
freqRange = [];
numFreq = 0;
phaseSeed = 0;
waveDir = 0;
waveSpread = 1;
wavegauge1loc = [0,0];
wavegauge2loc = [0,0];
wavegauge3loc = [0,0];
wF = [];
dw = [];
waveAmpTime = [];
waveAmpTime1 = [];
waveAmpTime2 = [];
waveAmpTime3 = [];
etaDataFile = 'etaData.mat';

data = importdata(etaDataFile) ;    % Import time-series
t = [0:dt:endTime]';      % WEC-Sim simulation time [s]
waveAmpTime = zeros(maxIt+1,2);
maxRampIT=round(rampTime/dt);
data_t = data(:,1)';                    % Data Time [s]
data_x = data(:,2)';                    % Wave Surface Elevation [m]
waveAmpTime(:,1) = t;
waveAmpTime(:,2) = interp1(data_t,data_x,t);
if rampTime~=0
    for i=1:maxRampIT
        waveAmpTime(i,2) = waveAmpTime(i,2)*(1+cos(pi+pi*(i-1)/maxRampIT))/2;
    end
end
waveAmpTime1        = zeros(maxIt+1,2);
waveAmpTime1(:,1)   = [0:maxIt]*dt;
waveAmpTime2        = zeros(maxIt+1,2);
waveAmpTime2(:,1)   = [0:maxIt]*dt;
waveAmpTime3        = zeros(maxIt+1,2);
waveAmpTime3(:,1)   = [0:maxIt]*dt;

File: source/objects_unittest/waveclass_unittest/testconfirm.py
import time

import waveClass_V3
waves = waveClass_V3.WaveClass('irregular')
bemFreq = [5.19999512307279,0.0199999977946844]
wDepth = 'infinite'
rampTime = 100
dt = 0.1
maxIt = 2000
g = 9.81
rho = 1000
endTime = 200
waves.freqRange = []
waves.freqDisc = 'EqualEnergy'
waves.numFreq = 500
waves.phaseSeed = 1
waves.wT = 8
waves.wH = 2.5
waves.spectrumType = 'BS'
waves.waveDir = [0]
waves.waveSpread = [1]
waves.wavegauge1loc = [0,0]
waves.wavegauge2loc = [0,0]
waves.wavegauge3loc = [0,0]
starttime = time.time()
waves.waveSetup(bemFreq,wDepth,rampTime,dt,maxIt,g, rho, endTime)
elapsed = time.time() - starttime
print(elapsed)
waves.plotEta(rampTime)
waves.plotEta()
waves.plotSpectrum()
